src/Binah/Frankie.hs

```diff
@@ -82,11 +82,12 @@ class HasSqlBackend config where
 backend :: (MonadConfig config m, HasSqlBackend config) => m Persist.SqlBackend
 backend = getSqlBackend <$> getConfig
 
-{-@ ignore httpAuthDb @-}
-{-@ assume httpAuthDb :: AuthMethod {v:(Entity User) | v == currentUser} (TaggedT<{\_ -> True}, {\_ -> False}> m)@-}
-httpAuthDb :: (MonadController w m, MonadConfig config m, HasSqlBackend config, MonadTIO m) => AuthMethod (Entity User) (TaggedT m)
-httpAuthDb = httpBasicAuth $ \username _password ->
-  mapTaggedT (reading backend) $ selectFirst (EntityFieldWrapper UserName ==. username)
+{-@ assume httpBasicAuth
+    :: (Username -> Password -> TaggedT<{\_ -> False}, {\_ -> False}> m (Maybe (Entity User)))
+    -> AuthMethod (Entity User) (TaggedT m)
```

Comment:
I'm not sure about the return type. I think it should be restricted for this to be sound.